lib/KDateRange/KDateCalendar.vue

```diff
@@ -0,0 +1,475 @@
+       * return true if date is last day of month for css border rounding
+       */
+      isLastDay(weekIndex, dayInWeekIndex, key, nextActiveMonthDay, nextActiveMonthDate) {
+        const result = this.getDayCell(
+          weekIndex,
+          dayInWeekIndex,
+          nextActiveMonthDay,
+          nextActiveMonthDate
+        );
+        let lastDay = null;
+        if (key === 'first') {
+          lastDay = new Date(this.activeYearStart, this.activeMonth + 1, 0).getDate();
+        } else {
+          lastDay = new Date(this.activeYearEnd, this.nextActiveMonth + 1, 0).getDate();
+        }
+        return result === lastDay;
+      },
+      goPrevMonth() {
+        const prevMonth = new Date(this.activeYearStart, this.activeMonth, 0);
+        this.activeMonth = prevMonth.getMonth();
+        this.activeYearStart = prevMonth.getFullYear();
+        this.activeYearEnd = prevMonth.getFullYear();
+      },
+      goNextMonth() {
+        const nextMonth = new Date(this.activeYearEnd, this.nextActiveMonth, 2);
+        this.activeMonth = nextMonth.getMonth();
+        this.activeYearStart = nextMonth.getFullYear();
+        this.activeYearEnd = nextMonth.getFullYear();
+      },
+      isValidDate(date) {
+        return new Date(date) !== 'Invalid Date' && !isNaN(new Date(date));
+      },
+      getDate(key, day) {
+        let currDate = null;
+        if (key === 'first') {
+          currDate = new Date(this.activeYearStart, this.activeMonth, day);
+        } else {
+          currDate = new Date(this.activeYearEnd, this.nextActiveMonth, day);
+        }
+        return currDate;
+      },
+      /**
+       * takes in index of month, returns string of month based on languagel/locale tag
+       */
+      monthString(monthIndex) {
+        const date = new Date();
+        date.setMonth(monthIndex);
+        return date.toLocaleString(this.dateLocale, { month: 'long' });
+      },
+    },
+  };
+
+</script>
+
+
+<style lang="css" scoped>
+
+  .calendar-wrap {
+    position: relative;
+  }
+
+  .left {
+    position: absolute;
+    top: 0;
+    left: 0;
+  }
+
+  .right {
+    position: absolute;
+    top: 0;
+    right: 0;
+  }
+
+  .months-text {
+    margin-bottom: 10px;
+    text-align: center;
+  }
+
+  .months-text .left {
+    float: left;
+    cursor: pointer;
+  }
+
+  .months-text .right {
+    float: right;
+    cursor: pointer;
+  }
+
+  .calendar {
+    height: auto;
+    margin-right: 5px;
+    font-family: 'Noto Sans';
+    font-size: 14px;
+    background: white;
+  }
+
+  .calendar ul {
+    list-style-type: none;
+  }
+
+  .calendar-month-left,
+  .calendar-month-right {
+    float: left;
+    width: 48%;
+  }
+
+  .calendar-month-right {
+    margin-left: 15px;
+  }
+
+  .calendar-days {
+    padding: 0;
+    margin: 0;
+  }
+
+  .calendar-days li {
+    display: inline-block;
+    width: 14%;
+    margin-top: 6px;
+  }
+
+  .calendar-days li.calendar-days--disabled {
+    color: var(--disabled-button-color);
```

Comment:
Do we use CSS vars elsewhere in KDS? Seems there [isn't support for them in IE 11](https://caniuse.com/?search=css%20vars)

Reply:
Just to affirm here that we can't use CSS vars in KDS, because of the lack of IE11 support. We've investigated a ponyfill, but not implemented it yet, due to the technical complexity (and because it is not a complete polyfill it comes with a bunch of caveats).

In general, using computed `:style` attributes is preferable, as it is a little simpler to reason about, but (especially for pseudo-selectors, which cannot be applied via `style` attributes) the `$computedClass` helper function lets you generate arbitrary CSS in JS: https://design-system.learningequality.org/colors/#usage

Reply:
Thank you @marcellamaki and @rtibbles!